### ISSUE TITLE
Book cover has strange flow and cut into 2 pages

### ISSUE THREAD

**Issue by iaomw** [readium/readium-css#50]
When using RLT css, this book cover is cut off and has strange flow.
Please check this link for testing files. https://github.com/readium/r2-testapp-swift/issues/6
As shown in the picture below, It could be reproduced with Safari on Mac. It has the same effect in our app using WKWebView. It should looks good in WKWebView. 
This is a SVG node inside h1 node. If I move everything from h1 to body, it will be fixed. 
So, there might be some problem with h1 styles.
<img width="1230" alt="untitled 8" src="https://user-images.githubusercontent.com/2307692/40238481-984d4f5a-5ab3-11e8-891f-c282866aebf2.png">

**Comment by iaomw** [comment]
@JayPanoz

**Comment by JayPanoz** [comment]
Hmmm, I can see two issues in `<head>` at first sight: 

1. `ReadiumCSS-before.css` is loaded after all other stylesheets while it should come first (before authors’ stylesheet(s);
2. the ReadiumCSS stylesheet used are the ones for CJK in vertical writing (for RTL, they should be the one in the `RTL` folder, [which is this one](https://github.com/readium/readium-css/tree/master/css/dist/rtl))

Do you get the same result with the RTL stylesheets?

**Comment by JayPanoz** [comment]
Also, I would recommend dividing this sample into 2 because, in and of itself, it exactly is an edge case we agreed to not handle back in December 2017 as the EPUB spec provides **zero** guidance for it.

So it would be a better idea to isolate Japanese from Arabic/Hebrew.

**Comment by iaomw** [comment]
Hi, the order doesn't matter, it will have the same effect. I have double checked it in Safari.

This one in picture is modified by myself on Mac, so it has this order. Don't worry. The webpage generated by Streamer has correct order.

The book is using CJK vertical mainly, and this cover html is mark as `xml:lang="ja"` 
Logically, we should use CJK vertical. What is the reason to use others? 

I just checked the css file under rtl, it will have some difference but still been cut off.

**Comment by JayPanoz** [comment]
OK I’ll take a look and report back. 

But if we end up discovering we need to isolate `h1` wrapping svg or images from others, there’s nothing we can do in CSS to achieve that.

**Comment by iaomw** [comment]
Thanks. I also found if the h1 has a width:(some number smaller than the container): important! 
It will also work fine. But I am not CSS expert. You guys should has better solutions.

**Comment by JayPanoz** [comment]
OK so I was able to reproduce this issue… without loading any ReadiumCSS stylesheet. 

![capture d ecran 2018-05-18 a 17 38 52](https://user-images.githubusercontent.com/12599652/40244601-8a3b3098-5ac3-11e8-8d59-62fc821121fe.png)

`h1` indeed has no dimensions set (`width` and `height`) so it can’t constrain the nested SVG which is `100%` of… nothing. This explains why putting a `width` on `h1` works, as you reported in the comment above.

With ReadiumCSS Stylesheets loaded we got this (note columns are on the `y-axis`).

![capture d ecran 2018-05-18 a 17 35 29](https://user-images.githubusercontent.com/12599652/40245096-e721a7b4-5ac4-11e8-9a41-dd95a4136b65.png)

![capture d ecran 2018-05-18 a 17 35 36](https://user-images.githubusercontent.com/12599652/40245100-ea0ed622-5ac4-11e8-9b5d-f6f005c190ae.png)

I’m reluctant to safeguard `h1` there, as some may well be more than the width of the page so our best option is probably [switching to viewport width (`vw`) there](https://github.com/readium/readium-css/blob/75747d553079e46da3ec4cc3e1b95bf0812d4268/css/src/modules/ReadiumCSS-safeguards-vertical.css#L58). But that is not a perfect solution as it will indeed create a blank page after.

I’ll try deploying the change on the `develop` branch ASAP so that we can see if it is good enough. But there’s not much we can do considering it doesn’t even work properly when kept untouched (and there’s consequently an authoring issue).
